### PR TITLE
basilk: init at 0.2.0

### PR DESCRIPTION
**Pull request by thtrf**
A Terminal User Interface (TUI) to manage your tasks with minimal kanban logic.

[repo](https://github.com/GabAlpha/basilk)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ba/basilk/package.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  src = fetchFromGitHub {
+    owner = "gabalpha";
+    repo = pname;
+    rev = "refs/tags/${version}";
+    sha256 = "sha256-b77vVf+WbDNzKwGaMJvgMEMCC41h5TXmg45OM9g4v+4=";
+  };
+
+  cargoHash = "sha256-ZYyRUqWbwAoNaGuCXJvqWTgUm5jNpihqvPe5SsfjEq4=";
+
+  meta = with lib; {
```
Comment (genga898) [inline pkgs/by-name/ba/basilk/package.nix:20]:
```suggestion
  meta =  {
```

File: pkgs/by-name/ba/basilk/package.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    description = "Terminal User Interface (TUI) to manage your tasks with minimal kanban logic";
+    homepage = "https://github.com/gabalpha/basilk";
+    license = licenses.asl20;
```
Comment (genga898) [inline pkgs/by-name/ba/basilk/package.nix:23]:
```suggestion
    license = lib.licenses.asl20;
```

File: pkgs/by-name/ba/basilk/package.nix [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    maintainers = [ maintainers.reecertv ];
```
Comment (genga898) [inline pkgs/by-name/ba/basilk/package.nix:24]:
```suggestion
    maintainers = with lib.maintainers; [ reecertv ];
```

**Review by genga898 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ba/basilk/package.nix (hunk 1 above)
Comment (genga898) [reply pkgs/by-name/ba/basilk/package.nix:20]:
```suggestion
  meta = {
```
Remove the with lib as it is only reference in two places within meta

File: pkgs/by-name/ba/basilk/package.nix (hunk 2 above)
Comment (genga898) [reply pkgs/by-name/ba/basilk/package.nix:23]:
```suggestion
    license = lib.licenses.asl20;
```
Replaces the `with lib;` statement  that was removed from the meta

**Comment by genga898** [comment]
Result of `nixpkgs-review pr 350812` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>basilk</li>
  </ul>
</details>

**Comment by Aleksanaa** [comment]
I unresolved the above two request changes by @genga898  before because they were not resolved. Do you have any questions or objections to these two changes?

**Comment by thtrf** [comment]
> I unresolved the above two request changes by @genga898 before because they were not resolved. Do you have any questions or objections to these two changes?

I prefer to use `with lib` because it avoids the need to explicitly reference `lib.` in multiple places.

**Review by genga898 — APPROVED** [review]

**Review by GaetanLepage — CHANGES_REQUESTED** [review]
A few minor requests.

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Pull request merged**